generic_server.cc

```diff
@@ -139,7 +140,13 @@ future<> connection::shutdown()
 server::server(const sstring& server_name, logging::logger& logger, const db::config& db_cfg)
     : _server_name{server_name}
     , _logger{logger}
+    , _conns_cpu_concurrency(db_cfg.uninitialized_connections_semaphore_cpu_concurrency)
 {
+    auto setup_conns_sem = [this] (const unsigned int &concurrency) {
+        _conns_cpu_concurrency_semaphore = make_lw_shared<named_semaphore>(concurrency, named_semaphore_exception_factory{"connections cpu concurrency semaphore"});
+    };
+    setup_conns_sem(_conns_cpu_concurrency);
+    _conns_cpu_concurrency.observe(setup_conns_sem);
```

Comment:
Ah, you replace the semaphore on update.

Isn't it simpler to add/remove units as necessary?

Reply:
yes, I will change, when writing this I didn't realize semaphore can go negative